modules/combined/test/tests/gap_heat_transfer_mortar/finite-2d/finite_rr.i

```diff
@@ -0,0 +1,353 @@
+  patch_size = 80
+  patch_update_strategy = auto
+  [plank]
+    type = GeneratedMeshGenerator
+    dim = 2
+    xmin = -0.3
+    xmax = 0.3
+    ymin = -10
+    ymax = 10
+    nx = 2
+    ny = 67
+    elem_type = ${elem}
+    boundary_name_prefix = plank
+  []
+  [plank_id]
+    type = SubdomainIDGenerator
+    input = plank
+    subdomain_id = 1
+  []
+
+  [block]
+    type = GeneratedMeshGenerator
+    dim = 2
+    xmin = 0.31
+    xmax = 0.91
+    ymin = 7.7
+    ymax = 8.5
+    nx = 3
+    ny = 4
+    elem_type = ${elem}
+    boundary_name_prefix = block
+    boundary_id_offset = 10
+  []
+  [block_id]
+    type = SubdomainIDGenerator
+    input = block
+    subdomain_id = 2
+  []
+
+  [combined]
+    type = MeshCollectionGenerator
+    inputs = 'plank_id block_id'
+  []
+  [block_rename]
+    type = RenameBlockGenerator
+    input = combined
+    old_block = '1 2'
+    new_block = 'plank block'
+  []
+
+  [secondary]
+    input = block_rename
+    type = LowerDBlockFromSidesetGenerator
+    sidesets = 'block_left'
+    new_block_id = '30'
+    new_block_name = 'frictionless_secondary_subdomain'
+  []
+  [primary]
+    input = secondary
+    type = LowerDBlockFromSidesetGenerator
+    sidesets = 'plank_right'
+    new_block_id = '20'
+    new_block_name = 'frictionless_primary_subdomain'
+  []
+[]
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+[]
+
+[Problem]
+  type = ReferenceResidualProblem
+  reference_vector = 'ref'
+  extra_tag_vectors = 'ref'
+  converge_on = 'disp_x disp_y temp'
+[]
+
+[Variables]
+  [disp_x]
+    order = ${order}
+    block = 'plank block'
+    scaling = ${fparse 2.0 / (E_plank + E_block)}
+  []
+  [disp_y]
+    order = ${order}
+    block = 'plank block'
+    scaling = ${fparse 2.0 / (E_plank + E_block)}
+  []
+  [temp]
+    order = ${order}
+    block = 'plank block'
+    scaling = 1e-1
+  []
+  [thermal_lm]
+    order = ${order}
+    block = 'frictionless_secondary_subdomain'
+    scaling = 1e-7
+  []
+  [frictionless_normal_lm]
+    order = ${order}
+    block = 'frictionless_secondary_subdomain'
+    use_dual = true
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [action]
+    generate_output = 'stress_xx stress_yy stress_zz vonmises_stress hydrostatic_stress strain_xx strain_yy strain_zz'
+    block = 'plank block'
+    use_automatic_differentiation = true
+    strain = FINITE
+    extra_vector_tags = 'ref'
+  []
+[]
+
+[Kernels]
+  [hc]
+    type = ADHeatConduction
+    variable = temp
+    use_displaced_mesh = true
+    block = 'plank block'
+    extra_vector_tags = 'ref'
+  []
+[]
+
+[Constraints]
+  [weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = plank_right
+    secondary_boundary = block_left
+    primary_subdomain = frictionless_primary_subdomain
+    secondary_subdomain = frictionless_secondary_subdomain
+    variable = frictionless_normal_lm
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+    extra_vector_tags = 'ref'
```

Comment:
Did you try not passing extra vector tags for this particular object? This constraint residual is truly a constraint residual, i.e. the contact constraint, in the Lagrange multiplier. I am not fully sure of what it's doing.

Reply:
Yeah, this was on accident it was still in here, good catch.